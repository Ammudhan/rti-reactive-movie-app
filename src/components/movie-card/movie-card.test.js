import React from "react";
import renderer, { create } from "react-test-renderer";
import MovieCard from "./movie-card";
import MovieService from "../../services/movieService";

describe("Movie Card Component", () => {

    it("should define the state as expected", async () => {
        const movie = {
            id: 429617,
            original_title: "Spider-Man: Far from Home",
            poster_path: "/rjbNpRMoVvqHmhmksbokcyCr7wn.jpg",
            overview: "Peter Parker and his friends go on a summer trip to Europe. However, they will hardly be able to rest - Peter will have to agree to help Nick Fury uncover the mystery of creatures that cause natural disasters and destruction throughout the continent.",
            release_date: "2019-06-28",
            title: "Spider-Man: Far from Home",
            vote_average: 7.8
        };
        const props = {
            location: {
                state: {
                    movieId: 429617
                }
            }
        };
        const mockData = {
            data: movie
        };
        const mock = jest.spyOn(MovieService, "getMovieById");
        const promise = Promise.resolve(mockData);
        mock.mockImplementation(() => promise);
        const component = create(<MovieCard {...props}/>);
        await promise;
        expect(mock).toHaveBeenCalled();
        const instance = component.getInstance();
        expect(instance.state.movie).toBe(movie);
        expect(instance.state.error).toBe('');
    });

    it('should matches snapshot', async () => {
        const movie = {
            id: 429617,
            original_title: "Spider-Man: Far from Home",
            poster_path: "/rjbNpRMoVvqHmhmksbokcyCr7wn.jpg",
            overview: "Peter Parker and his friends go on a summer trip to Europe. However, they will hardly be able to rest - Peter will have to agree to help Nick Fury uncover the mystery of creatures that cause natural disasters and destruction throughout the continent.",
            release_date: "2019-06-28",
            title: "Spider-Man: Far from Home",
            vote_average: 7.8
        };
        const props = {
            location: {
                state: {
                    movieId: 429617
                }
            }
        };
        const mockData = {
            data: movie
        };
        const mock = jest.spyOn(MovieService, "getMovieById");
        const promise = Promise.resolve(mockData);
        mock.mockImplementation(() => promise);

        const component = renderer.create(<MovieCard {...props}/>);
        await promise;

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

});