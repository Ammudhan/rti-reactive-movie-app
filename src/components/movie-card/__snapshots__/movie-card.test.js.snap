// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Card Component should matches snapshot 1`] = `
<span
  className="row"
>
  <span
    className="col-12 card-layout card-title d-flex align-items-center justify-content-start"
  >
    Spider-Man: Far from Home
  </span>
  <span
    className="col-5 col-md-4 offset-1 "
  >
    <img
      alt="Spider-Man: Far from Home"
      className="img-fluid link"
      src="https://image.tmdb.org/t/p/w500/rjbNpRMoVvqHmhmksbokcyCr7wn.jpg"
    />
  </span>
  <span
    className="col-5 offset-md-1 card-info"
  >
    <div>
      2019
    </div>
    <div>
       mins
    </div>
    <div>
      7.8
       / 10
    </div>
    <div>
      <button
        className="favorite-button"
      >
        Mark As Favorite
      </button>
    </div>
    <div
      className="overview-desktop d-none d-md-block"
    >
      Peter Parker and his friends go on a summer trip to Europe. However, they will hardly be able to rest - Peter will have to agree to help Nick Fury uncover the mystery of creatures that cause natural disasters and destruction throughout the continent.
    </div>
  </span>
  <span
    className="col-12 d-none d-md-block mb-5"
  />
  <span
    className="col-12 d-md-none card-layout overview d-flex align-items-center"
  >
    Peter Parker and his friends go on a summer trip to Europe. However, they will hardly be able to rest - Peter will have to agree to help Nick Fury uncover the mystery of creatures that cause natural disasters and destruction throughout the continent.
  </span>
</span>
`;
